frontend/packages/process-editor/src/classes/ProcessTaskManager/RemoveProcessTaskManager.ts

```diff
@@ -0,0 +1,77 @@
+import { type BpmnApiContextProps } from '../../contexts/BpmnApiContext';
+import { type BpmnDetails } from '../../types/BpmnDetails';
+import { type TaskEvent } from './types';
+import { BpmnTypeEnum } from '../../enum/BpmnTypeEnum';
+import {
+  getDataTypeIdFromBusinessObject,
+  getLayoutSetIdFromTaskId,
+} from '../../utils/hookUtils/hookUtils';
+import { type Policy } from '../../utils/policy/types';
+import { PaymentPolicyBuilder } from '../../utils/policy';
+
+export class RemoveProcessTaskManager {
+  constructor(
+    private readonly org: string,
+    private readonly app: string,
+    private readonly layoutSets: BpmnApiContextProps['layoutSets'],
+    private readonly deleteLayoutSet: BpmnApiContextProps['deleteLayoutSet'],
+    private readonly deleteDataTypeFromAppMetadata: BpmnApiContextProps['deleteDataTypeFromAppMetadata'],
+    private readonly bpmnDetails: BpmnDetails,
+    private readonly currentPolicy: Policy,
+  ) {}
+
+  public handleTaskRemove(taskEvent: TaskEvent): void {
+    if (this.bpmnDetails.type === BpmnTypeEnum.Task) {
+      this.handleDataTaskRemove();
+    }
+
+    if (this.bpmnDetails.taskType === 'payment') {
+      this.handlePaymentTaskRemove(taskEvent);
+    }
+
+    if (this.bpmnDetails.taskType === 'signing') {
+      this.handleSigningTaskRemove(taskEvent);
+    }
+  }
+
+  private handleDataTaskRemove(): void {
+    const layoutSetId = getLayoutSetIdFromTaskId(this.bpmnDetails, this.layoutSets);
+
+    if (layoutSetId) {
+      this.deleteLayoutSet({
+        layoutSetIdToUpdate: layoutSetId,
+      });
+    }
+  }
+
+  private handlePaymentTaskRemove(taskEvent: TaskEvent): void {
```

Comment:
Are we missing a couple of lines deleting the payment layoutset here?

Reply:
Let me look into that. 💯

Reply:
Nice catch!